### PR TITLE
Added handler for ZIO effects

### PR DESCRIPTION
**Pull request by sksamuel**
If you're happy with this Greg I'll merge and then release 2.4.4

### CODE REVIEW

File: pulsar4s-zio/src/test/scala/com/sksamuel/pulsar4s/zio/ZioAsyncHandlerTest.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  implicit val schema: Schema[String] = Schema.STRING
+
+  val client = PulsarClient("pulsar://localhost:6650")
+  val topic = Topic("persistent://sample/standalone/ns1/monix_" + UUID.randomUUID())
+
+  override def afterAll(): Unit = {
+    client.close()
+  }
+
+  test("async producer should use monix") {
```
Comment (gmethvin) [inline pulsar4s-zio/src/test/scala/com/sksamuel/pulsar4s/zio/ZioAsyncHandlerTest.scala]: Should be zio not monix

Comment (sksamuel) [reply pulsar4s-zio/src/test/scala/com/sksamuel/pulsar4s/zio/ZioAsyncHandlerTest.scala]: Done.

**Comment by cipriansofronia** [comment]
@sksamuel, may I change some stuff here? :angel:

**Comment by sksamuel** [comment]
Feel free.

**Comment by cipriansofronia** [comment]
I can't push to this repo, could you grant me some permissions to push to this branch?

**Comment by sksamuel** [comment]
Try now.

**Comment by cipriansofronia** [comment]
Pushed! :tada:  Let me know what you think :wink:

**Comment by sksamuel** [comment]
Can you explain what you changed briefly ?

**Comment by cipriansofronia** [comment]
> Can you explain what you changed briefly ?

So, basically, you don't need to transform from java Future to scala Future and then transform it again to ZIO, ZIO has a lib that would convert java futures to ZIO effortless, and that was it, after that I just applied this conversion to every function from `pulsar` api

**Comment by sksamuel** [comment]
I see thanks!

**Pull request merged**